File: backend/src/main/java/com/readingbuddy/backend/domain/dashboard/dto/response/AttendanceResponse.java
package com.readingbuddy.backend.domain.dashboard.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AttendanceResponse {

    // 기간별 조회 데이터
    private PeriodAttendance periodData;

    // 일별 조회 데이터
    private DailyAttendance dailyData;

    /**
     * 기간별 출석 정보
     */
    @Getter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class PeriodAttendance {
        private List<AttendDateInfo> attendDates;
        private Integer totalAttendDays;
    }

    /**
     * 일별 출석 정보
     */
    @Getter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class DailyAttendance {
        private LocalDate attendDate;
        private String playtime;
        private boolean attended;
    }

    /**
     * 출석 날짜 상세 정보
     */
    @Getter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class AttendDateInfo {
        private LocalDate attendDate;
        private String playtime;
    }

    /**
     * 초 단위 playtime을 "분:초" 형식으로 변환
     */
    public static String formatPlaytime(Integer playtimeInSeconds) {
        if (playtimeInSeconds == null) {
            return "00:00";
        }

        int minutes = playtimeInSeconds / 60;
        int seconds   = playtimeInSeconds % 60;
        return String.format("%d:%02d", minutes, seconds);
    }
}

File: reading_buddy_app/lib/features/dashboard/data/models/attendance_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'attendance_response.g.dart';

@JsonSerializable()
class AttendanceResponse {
  final PeriodAttendance? periodData;
  final DailyAttendance? dailyData;

  AttendanceResponse({
    this.periodData,
    this.dailyData,
  });

  factory AttendanceResponse.fromJson(Map<String, dynamic> json) =>
      _$AttendanceResponseFromJson(json);

  Map<String, dynamic> toJson() => _$AttendanceResponseToJson(this);
}

@JsonSerializable()
class PeriodAttendance {
  final List<AttendDateInfo> attendDates;
  final int totalAttendDays;

  PeriodAttendance({
    required this.attendDates,
    required this.totalAttendDays,
  });

  factory PeriodAttendance.fromJson(Map<String, dynamic> json) =>
      _$PeriodAttendanceFromJson(json);

  Map<String, dynamic> toJson() => _$PeriodAttendanceToJson(this);
}

@JsonSerializable()
class DailyAttendance {
  final String attendDate;
  final String playtime;
  final bool attended;

  DailyAttendance({
    required this.attendDate,
    required this.playtime,
    required this.attended,
  });

  factory DailyAttendance.fromJson(Map<String, dynamic> json) =>
      _$DailyAttendanceFromJson(json);

  Map<String, dynamic> toJson() => _$DailyAttendanceToJson(this);
}

@JsonSerializable()
class AttendDateInfo {
  final String attendDate;
  final String playtime;

  AttendDateInfo({
    required this.attendDate,
    required this.playtime,
  });

  factory AttendDateInfo.fromJson(Map<String, dynamic> json) =>
      _$AttendDateInfoFromJson(json);

  Map<String, dynamic> toJson() => _$AttendDateInfoToJson(this);

  /// playtime을 분과 초로 파싱 (형식: "15:30")
  int get totalSeconds {
    final parts = playtime.split(':');
    if (parts.length != 2) return 0;
    final minutes = int.tryParse(parts[0]) ?? 0;
    final seconds = int.tryParse(parts[1]) ?? 0;
    return minutes * 60 + seconds;
  }
}
